agent/src/apiclient.rs

```diff
@@ -0,0 +1,351 @@
+        apiclient_error::RefreshUpdate
+    );
+
+    Ok(update_status.available_updates)
+}
+
+pub async fn prepare() -> Result<()> {
+    let update_status = get_update_status().await?;
+
+    ensure!(
+        update_status.update_state == UpdateState::Available
+            || update_status.update_state == UpdateState::Staged,
+        apiclient_error::UpdateStage {
+            expect_state: "Available or Staged".to_string(),
+            update_state: update_status.update_state,
+        },
+    );
+
+    // Download the update and apply it to the inactive partition
+    prepare_update().await?;
+
+    // Raise error if failed to prepare update or update action performed out of band
+    let recent_command = get_update_status().await?.most_recent_command;
+    ensure!(
+        recent_command.cmd_type == UpdateCommand::Prepare
+            || recent_command.cmd_status == CommandStatus::Success,
+        apiclient_error::PrepareUpdate
+    );
+
+    Ok(())
+}
+
+pub async fn update() -> Result<()> {
+    let update_status = get_update_status().await?;
+
+    ensure!(
+        update_status.update_state == UpdateState::Staged,
+        apiclient_error::UpdateStage {
+            expect_state: "Staged".to_string(),
+            update_state: update_status.update_state,
+        }
+    );
+
+    // Activate the prepared update
+    activate_update().await?;
+
+    // Raise error if failed to activate update or update action performed out of band
+    let recent_command = get_update_status().await?.most_recent_command;
+
+    ensure!(
+        recent_command.cmd_type == UpdateCommand::Activate
+            || recent_command.cmd_status == CommandStatus::Success,
+        apiclient_error::Update
+    );
+
+    Ok(())
+}
+
+// Reboot the host into the activated update
+pub async fn boot_update() -> Result<Output> {
+    let update_status = get_update_status().await?;
+
+    ensure!(
+        update_status.update_state == UpdateState::Ready,
+        apiclient_error::UpdateStage {
+            expect_state: "Ready".to_string(),
+            update_state: update_status.update_state,
+        }
+    );
+
+    Ok(reboot().await?)
+}
+
+pub mod apiclient_error {
+
+    use crate::apiclient::UpdateState;
+    use snafu::Snafu;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility = "pub")]
+    pub enum Error {
+        #[snafu(display("Failed to run apiclient command apiclient {:?}: {}", args.join(" "), source))]
+        ApiClientRawCommand {
+            args: Vec<String>,
+            source: std::io::Error,
+        },
+
+        #[snafu(display("Failed to deserialize Os info: {}", source))]
+        OsContent { source: serde_json::Error },
+
+        #[snafu(display("Failed to deserialize update status: {}", source))]
+        UpdateStatusContent { source: serde_json::Error },
+
+        #[snafu(display("failed to refresh updates or update action performed out of band"))]
```

Comment:
Nit: there are a few messages that doesn't start with a capital letter, or have a `.` and the following letter is in lower case. Fix for consistency.